python/ray/tests/test_usage_stats.py

```diff
@@ -30,5 +86,108 @@ def test_usage_lib_cluster_metadata_generation(shutdown_only):
     )
 
 
+def test_usage_lib_report_data(shutdown_only, shutdown_serve):
```

Comment:
Can you add tests for different kinds of failures? E.g., GCS not responding via dependency injection. Let's make sure no error messages are logged by default.

Reply:
Lmk if you think this is not enough! https://github.com/ray-project/ray/pull/22249/commits/7134686cc8005452c169be7e4bd512adc0d0e870